docs/adr/0014-auth-for-settings-redesign.md

```diff
@@ -0,0 +1,87 @@
+# Authentication in Settings Redesign
+
+- Status: proposed
+- Deciders: Les Orchard
+- Date: 2020-04-14
+
+Issue: https://jira.mozilla.com/browse/FXA-1559
+
+## Context and Problem Statement
+
+The Settings Redesign app needs to query & mutate the same protected user data as the existing settings app hosted on content-server. This will require some form of authentication & authorization to manage that data.
+
+## Decision Drivers
+
+- Smooth UX
+- Security
+- Development velocity
+- Ease of integration
+
+## Considered Options
+
+- Share existing sessionToken to use auth-server APIs
+- Authenticate via OAuth2 to use auth-server APIs
+- Authenticate via OAuth2 to use DB directly
+- Implement own login sessions
+
+## Decision Outcome
+
+Chosen option: "Share existing sessionToken to use auth-server APIs", because it will have the least impact on existing auth-server implementation and infrastructure outside the Settings Redesign app in general.
+
+### Positive Consequences
+
+- The new Settings Redesign app can reuse existing auth-server APIs in the same way as existing settings code. This can constrain the novelty in the project to the redesigned & reimplemented settings UX.
+
+### Negative Consequences
+
+- Sharing the sessionToken from content-server to new Settings Redesign app will require a novel mechanism that will, in turn, require close security review.
+- We don't have an entirely greenfield project, which could be a bit of a drag. But, rediscovering lessons learned in existing code can also be a drag.
+
+## Pros and Cons of the Options
+
+### Share existing sessionToken to use auth-server APIs
+
+- Pros:
+  - Requires few or no auth-server changes - APIs already expect sessionToken
+  - Could be done with seamless UX - user visits new settings without confirmation
+- Neutral:
+  - GraphQL server needs to proxy auth-server APIs
+- Cons:
+  - Novel security properties - need to carefully convey sessionToken across origins
+  - Will need security review & some hard thinking about this implementation some prior art by shane for payment server that we never shipped
```

Comment:
~shane~ Shane, plus should there be a "this implementation _from_ some prior art" (would be nice to have a link as well)?

Reply:
Yeah, there should be a link, or I should remove this. Copypasta oopsie